cmd/oras/internal/option/remote.go

```diff
@@ -248,21 +249,30 @@ func (opts *Remote) Credential() auth.Credential {
 }
 
 // NewRegistry assembles a oras remote registry.
-func (opts *Remote) NewRegistry(hostname string, common Common) (reg *remote.Registry, err error) {
+func (opts *Remote) NewRegistry(hostname string, warn func(...interface{}), common Common) (reg *remote.Registry, err error) {
 	reg, err = remote.NewRegistry(hostname)
 	if err != nil {
 		return nil, err
 	}
 	hostname = reg.Reference.Registry
 	reg.PlainHTTP = opts.isPlainHttp(hostname)
+
+	if opts.distributionSpec.referrersAPI == nil || *opts.distributionSpec.referrersAPI {
```

Comment:
Why do we check Referrers API here? Can you help me to understand it?

Reply:
I assumed it was introduced after 1.1.0 so should be avoid if referrers API is explicitly specified as not supported. Looks like those two features are irrelevant.

Reply:
Updated